core/handlers/v2/simulation_views_v5.go

```diff
@@ -40,30 +40,43 @@ type MatcherViewV5 struct {
 	Config  map[string]interface{} `json:"config,omitempty"`
 }
 
+type LiteralViewV5 struct {
+	Name  string      `json:"name"`
+	Value interface{} `json:"value"`
+}
+
+type VariableViewV5 struct {
+	Name       string `json:"name"`
+	Method     string `json:"method"`
+	Expression string `json:"expression"`
+}
+
 func NewMatcherView(matcher string, value interface{}) MatcherViewV5 {
 	return MatcherViewV5{
 		Matcher: matcher,
 		Value:   value,
 	}
 }
 
-//Gets Response - required for interfaces.RequestResponsePairView
+// Gets Response - required for interfaces.RequestResponsePairView
 func (this RequestMatcherResponsePairViewV5) GetResponse() interfaces.Response { return this.Response }
 
 type ResponseDetailsViewV5 struct {
 	Status           int                    `json:"status"`
 	Body             string                 `json:"body"`
-	BodyFile         string              	`json:"bodyFile,omitempty"`
+	BodyFile         string                 `json:"bodyFile,omitempty"`
 	EncodedBody      bool                   `json:"encodedBody"`
 	Headers          map[string][]string    `json:"headers,omitempty"`
 	Templated        bool                   `json:"templated"`
 	TransitionsState map[string]string      `json:"transitionsState,omitempty"`
 	RemovesState     []string               `json:"removesState,omitempty"`
 	FixedDelay       int                    `json:"fixedDelay,omitempty"`
 	LogNormalDelay   *LogNormalDelayOptions `json:"logNormalDelay,omitempty"`
+	Literals         []LiteralViewV5        `json:"literals,omitempty"`
+	Variables        []VariableViewV5       `json:"variables,omitempty"`
```

Comment:
Do we want to apply this for a particular response only?  I thought the idea of having a custom variable is that we can declare a template function globally and use it across multiple responses. This will give you some benefits in terms of managing templating data for a large simulation file, like you don't need to go through each response to modify the function when change is required.

For example, the `DataViewV5` could be changed to: 
```
type DataViewV5 struct {
	RequestResponsePairs []RequestMatcherResponsePairViewV5 `json:"pairs"`
	GlobalActions        GlobalActionsView                  `json:"globalActions"`
	GlobalVariables     GlobalVariablesView               `json:"globalVariables"`

}
```
where the `GlobalVariables` will be the root for the literals and the variables.

Reply:
@tommysitu I was with assumption that he wants to have per request/response as mentioned about long response with long xpath/json path expression and he wants to replace multiple occurrences in a response. Yea, in that case we can have globally with func and args.